src/data_access/FirebaseUserDataAccessObject.java

```diff
@@ -17,22 +21,93 @@ public class FirebaseUserDataAccessObject
   static OkHttpClient client = new OkHttpClient();
 
   @Override
-  public User get(String email) {
-    Request request =
-        new Request.Builder()
-            .url(
-                Constants.FIREBASE_URL
-                    + "users/"
-                    + email
-                    + ".json") // TODO: Ensure this successfully searches by email
-            .method("GET", null)
-            .build();
+  public Optional<User> get(String email, String password) {
+    // Authentication Request
+    JSONObject jsonBody = new JSONObject();
+    jsonBody.put("email", email);
+    jsonBody.put("password", password);
+    jsonBody.put("returnSecureToken", true);
+
+    MediaType JSON = MediaType.parse("application/json; charset=utf-8");
+    RequestBody body = RequestBody.create(JSON, jsonBody.toString());
+
+    String authUrl =
+        "https://identitytoolkit.googleapis.com/v1/accounts:signInWithPassword?key=AIzaSyBuJk14Gljk-chdN_9YVywSKnf38ttwUVg";
+
+    Request authRequest = new Request.Builder().url(authUrl).post(body).build();
+
+    try {
+      Response authResponse = client.newCall(authRequest).execute();
+      String authResponseData = authResponse.body().string();
+
+      if (!authResponse.isSuccessful()) {
+        System.out.println("Authentication failed: " + authResponseData);
+        return Optional.empty();
+      }
+
+      JSONObject authResponseJson = new JSONObject(authResponseData);
+
+      if (!authResponseJson.has("idToken") || !authResponseJson.has("localId")) {
+        System.out.println("Invalid authentication response");
+        return Optional.empty();
+      }
+
+      String idToken = authResponseJson.getString("idToken");
+
+      return getUserData(idToken)
+          .map(
+              userData -> {
+                String displayName = userData[0];
+                long createdAt = Long.parseLong(userData[1]);
+                Instant instant = Instant.ofEpochMilli(createdAt);
+                LocalDateTime dateTime = LocalDateTime.ofInstant(instant, ZoneId.systemDefault());
+
+                System.out.println("Display Name: " + displayName);
+                System.out.println("Created At: " + dateTime);
+
+                CommonUserFactory userFactory = new CommonUserFactory();
+                return userFactory.create(email, displayName, password, dateTime);
+              });
+    } catch (IOException | JSONException e) {
+      System.out.println("Error during authentication: " + e.getMessage());
+      return Optional.empty();
+    }
+  }
+
+  // Return username, time user was created
+  public Optional<String[]> getUserData(String idToken) {
```

Comment:
This is not good coding practice, we should not be returning a String array.
Is there a way to return a User object and allow the Interactor code to extract the relevant User data?

Reply:
Yes, this will be replaced by it's own class eventually. Surprisingly, String array is the cleanest way to do it rn.

Reply:
I corrected this to make the method private for now.

Reply:
Thanks for making this private.